ionics_fits/common.py

```diff
@@ -852,6 +859,17 @@ def free_func(x: TX, *free_param_values: float) -> TY:
             for param, value in uncertainties.items()
         }
 
+        # rescale the covariance matrix
+        free_param_scales = np.array(
+            [
+                self.model.parameters[param].scale_factor
+                for param in self.free_parameters
+            ]
+        )
+        covariance_scales = np.tile(free_param_scales, (len(self.free_parameters), 1))
+        covariance_scales = np.multiply(covariance_scales, covariance_scales.T)
+        self.covariances = np.multiply(covariances, covariance_scales)
```

Comment:
For my own benefit....
This code is equivalent to
```python
free_param_scales = np.array(
            [
                self.model.parameters[param].scale_factor
                for param in self.free_parameters
            ]
        )[:, np.newaxis]
self.covariances = free_param_scales * free_param_scales.T * covariances
```

Reply:
It would make sense for us to divide by the scale of a parameter before doing floating point optimisations...

Therefore we would need to multiply the scale back on. Variance scales quadratically with the parameter, so this passes a basic sense check.